sources/constraint-declaration-validation.asciidoc

```diff
@@ -841,16 +845,30 @@ In the example above,
 * for each extracted list of addresses, the `@NotEmpty` constraint will be validated and the extractor for `List` elements will be invoked, providing the `Address` objects from each list in the map
 * the `@ValidAddress` constraint will be applied to all elements of all lists stored in the map
 
-[tck-not-testable]#Container element constraints are not supported on wildcard type arguments (with or without bounds), i.e. the following usage is unsupported#:
+[tck-not-testable]#It is not supported to declare container element constraints on the type parameters of generic types or methods.#
+[tck-not-testable]#It is also not supported to declare container element constraints on type arguments within the `extends` or `implements` clauses of type definitions.#
+I.e. the following usages are unsupported:
 
-.Unsupported usage of a container element constraint on a wildcard type argument
+.Unsupported usage of container element constraints on generic types and methods
 ====
 [source, JAVA]
 ----
-private List<@NotEmpty ?> list;
+public class NonNullList<@NotNull T> {
+    [...]
+}
+
+public class ContainerFactory {
+    <@NotNull T> Container<T> instantiateContainer(Class<T> clazz) { [...] }
```

Comment:
I would point out here that Class is perhaps a bad example as there isn't any meaningful way to extract instances of its type parameter.